cmd/link.go

```diff
@@ -50,16 +52,34 @@ Please perform 'odo service create %s ...' before attempting to link the service
 			os.Exit(1)
 		}
 
-		// we also need to check whether there is a secret with the same name as the service
-		// the secret should have been created along with the secret
-		_, err = client.GetSecret(serviceName, projectName)
+		// if there is a ServiceBinding, then that means there is a secret (or perhaps there will be soon)
+		// which we can link to
+		_, err = client.GetServiceBinding(serviceName, projectName)
 		if err != nil {
-			fmt.Printf(`Secret %s should have been created along with the service
-If you previously created the service with 'odo service create', then you may have to wait a few seconds until OpenShift provisions it.
-If not, then please delete the service and recreate it using 'odo service create %s`, serviceName, serviceName)
+			fmt.Printf(`The service was not created via odo. 
+Please delete service and recreate it using 'odo service create %s'`, serviceName)
 			os.Exit(1)
 		}
 
+		if wait {
+			// we wait until the secret has been created on the OpenShift
+			// this is done because the secret is only created after the Pod that runs the
+			// service is in running state.
+			// This can take a long time to occur if the image of the service has yet to be downloaded
```

Comment:
It might be good to show a message here. Something like "Waiting for %s service to come up."

Reply:
Will do that ASAP

Reply:
Fixed